File: 1-easy/00058-length-of-last-word/solution.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {

        int count = 0;
        bool is_counting_started = false;

        while (!s.empty()) {
            char ch = s.back();
            s.pop_back();
            if (ch != ' ') {
                is_counting_started = true;
            }

            if (is_counting_started) {
                if (ch != ' ') {
                    count += 1;
                } else {
                    break;
                }
            }
        }

        return count;
    }
};

File: 1-easy/00058-length-of-last-word/solution.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {

        let mut s_bytes = s.into_bytes();
        let mut count = 0_i32;
        let mut is_counting_started = false;
    
        while let Some(byte) = s_bytes.pop() {
            if byte != b' ' {
                is_counting_started = true;
            }

            if is_counting_started {
                if byte != b' ' {
                    count += 1;
                } else {
                    break;
                }
            }
        }

        count
    }
}
